### PR TITLE
Amend "calc: new recipe for a C-style arbitrary precision calculator"

### PR DESCRIPTION
**Pull request by raefaldhia**
Oppss.

In current recipe we removed ldconfig due to there's no ldconfig, but i just found out that ldconfig is exist with name ld in haiku.

I found that from:
https://github.com/haikuports/haikuports/blob/75a3cf5384f650242f4b8a1094ab305d33ed99e7/app-emulation/pcem/pcem-10.1.recipe#L39

should i revert or make new pr?

But look like there's no big impact either we add it or not.

Amend haikuports/haikuports#901

### CODE REVIEW

**Comment by raefaldhia** [comment]
Or maybe you could do `git --amend` for me to make commit message clear.
There's just little difference, about `LDCONFIG=ln` and add `cmd:ld$secondaryArchSuffix`

File: sci-mathematics/calc/calc-2.12.5.3.recipe [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		HAVE_UNISTD_H=YES \
+		HAVE_FPOS_POS=1 \
+		HAVE_UID_T=1 \
+		HAVE_USTAT=1 \
+		EXTRA_CFLAGS="-DUSE_TERMIOS" \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		LN="ln -s" \
+		LDCONFIG=ld \
```
Comment (raefaldhia) [inline sci-mathematics/calc/calc-2.12.5.3.recipe]: Here

File: sci-mathematics/calc/calc-2.12.5.3.recipe [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	cmd:sed
+	cmd:diff
+	cmd:egrep
+	cmd:ctags
+	cmd:xargs
+	cmd:cmp
+	cmd:touch
+	cmd:less
+	cmd:strip
+	cmd:ld$secondaryArchSuffix
```
Comment (raefaldhia) [inline sci-mathematics/calc/calc-2.12.5.3.recipe]: Here

File: sci-mathematics/calc/calc-2.12.5.3.recipe [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		HAVE_UNISTD_H=YES \
+		HAVE_FPOS_POS=1 \
+		HAVE_UID_T=1 \
+		HAVE_USTAT=1 \
+		EXTRA_CFLAGS="-DUSE_TERMIOS" \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		LN="ln -s" \
+		LDCONFIG=ld \
```
Comment (raefaldhia) [inline sci-mathematics/calc/calc-2.12.5.3.recipe]: Here

**Comment by raefaldhia** [comment]
Closing this, i need to test.

**Comment by pulkomandy** [comment]
ldconfig and ld are not  the same thing.

ld is the linker, it is used to take a bunch of .o files (compiled by gcc) and make a library or executable from them.
ldconfig is a Linux specific tool, it is used to scan all libraries in the system (/usr/lib, /lib, ...) and generate a database/cache to map their soname to complete file paths.

On Haiku, there is no database/cache and the lookup is performed by opening the directories and looking for the files. We always make sure the soname matches the filename, which makes this not cost too much. As a result, ldconfig is not needed on Haiku.

**Comment by raefaldhia** [comment]
@pulkomandy ok then thanks, i thought it's same since when i typed --help ldconfig and ld look alike.

**Pull request closed without merging**